Tests/DispatchAsyncTests/DispatchGroupTests.swift

```diff
@@ -15,109 +15,188 @@
 @_spi(DispatchAsync) import DispatchAsync
 import Testing
 
+import func Foundation.sin
+
+#if !os(WASI)
+import class Foundation.Thread
+#endif
+
 private typealias DispatchGroup = DispatchAsync.DispatchGroup
+private typealias DispatchQueue = DispatchAsync.DispatchQueue
 
-@Test(arguments: [100])
-func dispatchGroupOrderCleanliness(repetitions: Int) async throws {
-    // Repeating this `repetitions` number of times to help rule out
-    // edge cases that only show up some of the time
-    for index in 0 ..< repetitions {
-        Task {
-            actor Result {
-                private(set) var value = ""
-
-                func append(value: String) {
-                    self.value.append(value)
+@Suite("DispatchGroup Tests")
+struct DispatchGroupTests {
+    @Test(arguments: [1000])
+	@available(macOS 13, iOS 16, tvOS 16, watchOS 9, *)
+	func dispatchGroupOrderCleanliness(repetitions: Int) async throws {
+    	// Repeating this `repetitions` number of times to help rule out
+	    // edge cases that only show up some of the time
+    	for index in 0 ..< repetitions {
+            Task {
+                actor Result {
+                    private(set) var value = ""
+
+                    func append(value: String) {
+                        self.value.append(value)
+                    }
                 }
-            }
 
-            let result = Result()
+                let result = Result()
 
-            let group = DispatchGroup()
-            await result.append(value: "|🔵\(index)")
+                let group = DispatchGroup()
+                await result.append(value: "|🔵\(iteration)")
```

Comment:
Ah, interesting. Compiler error caused by this change. Something in the rebase and merged didn't flag this as a conflict, and interestingly enough the CI didn't run on this PR after re-targeting the stacked PR to `main`. Will fix soon, and will check that a normal PR runs the expected checks.